rust/examples-frontend/svc/src/index.ts

```diff
@@ -0,0 +1,338 @@
+                        // which you should definitely have in real-world applications
+                        waitingForResponse.set('ConnectedProducerTransport', () =>
+                        {
+                            success();
+                            console.log('Producer transport connected');
+                        });
+                    })
+                    .on('produce', ({ kind, rtpParameters }, success) =>
+                    {
+                        // Once connection is established, send request to produce
+                        // audio or video track
+                        send({
+                            action : 'Produce',
+                            kind,
+                            rtpParameters,
+                        });
+                        // And wait for confirmation, but, obviously, no error handling,
+                        // which you should definitely have in real-world applications
+                        waitingForResponse.set('Produced', ({ id }: {id: string}) =>
+                        {
+                            success({ id });
+                        });
+                    });
+
+                // Request microphone and camera access, in real-world apps you may want
+                // to do this separately so that audio-only and video-only cases are
+                // handled nicely instead of failing completely
+                const mediaStream = await navigator.mediaDevices.getUserMedia({
+                    audio : true,
+                    video : {
+                        width : {
+                            ideal : 1280
+                        },
+                        height : {
+                            ideal : 720
+                        },
+                        frameRate : {
+                            ideal : 60
+                        }
+                    }
+                });
+
+                sendPreview.srcObject = mediaStream;
+
+                const producers = [];
+
+                // And create producers for all tracks that were previously requested
+                for (const track of mediaStream.getTracks()) {
+                    const codec = track.kind === 'video' ? device.rtpCapabilities.codecs?.find((codec) => codec.mimeType.toLowerCase() === 'video/vp9') ?? device.rtpCapabilities.codecs?.find((codec) => codec.mimeType.toLowerCase() === 'video/vp8') : undefined
+                    let encodings
+
+                    if (codec?.mimeType.toLowerCase() === 'video/vp8') {
+                        encodings = [
+                            { maxBitrate: 100000 },
+                            { maxBitrate: 300000 },
+                            { maxBitrate: 900000 },
+                        ]
+                    } else if (codec?.mimeType.toLowerCase() === 'video/vp9') {
+                        encodings = [
+                            { scalabilityMode: 'S3T3' },
+                        ]
+                    }
```

Comment:
Since this example is for both simulcast and SVC, should probably be called accordingly.

Reply:
agree